### PR TITLE
Syntax: highlight keyword "select"

### PR DESCRIPTION
**Pull request by sudo-nice**
Reference: https://github.com/crystal-lang/crystal/blob/master/spec/std/concurrent/select_spec.cr

### CODE REVIEW

File: syntax/crystal.vim [hunk 1]
```diff
@@ -230,11 +230,13 @@ if !exists('b:crystal_no_expensive') && !exists('g:crystal_no_expensive')
...
-  syn region crystalCaseExpression	       matchgroup=crystalConditional start="\<case\>"  end="\<end\>" contains=ALLBUT,@crystalNotTop fold
+  syn region crystalBlockExpression  matchgroup=crystalControl     start="\<begin\>"  end="\<end\>" contains=ALLBUT,@crystalNotTop fold
+  syn region crystalCaseExpression   matchgroup=crystalConditional start="\<case\>"   end="\<end\>" contains=ALLBUT,@crystalNotTop fold
+  syn region crystalSelectExpression matchgroup=crystalConditional start="\<select\>" end="\<end\>" contains=ALLBUT,@crystalNotTop fold
   syn region crystalConditionalExpression matchgroup=crystalConditional start="\%(\%(^\|\.\.\.\=\|[{:,;([<>~\*/%&^|+=-]\|\%(\<[_[:lower:]][_[:alnum:]]*\)\@<![?!]\)\s*\)\@<=\%(if\|ifdef\|unless\)\>" end="\%(\%(\%(\.\@<!\.\)\|::\)\s*\)\@<!\<end\>" contains=ALLBUT,@crystalNotTop fold
 
-  syn match crystalConditional "\<\%(then\|else\|when\)\>[?!]\@!"	contained containedin=crystalCaseExpression
+  syn match crystalConditional "\<\%(then\|else\|when\)\>[?!]\@!"  contained containedin=crystalCaseExpression
+  syn match crystalConditional "\<\%(when\)\>[?!]\@!"              contained containedin=crystalSelectExpression
```
Comment (rhysd) [inline syntax/crystal.vim]: As far as I saw the implementation, `select ... end` block seems to be able to take `else` for nonblocking. 

https://github.com/crystal-lang/crystal/blob/bec784dae649922a43932df4afd777cb83c1b107/src/compiler/crystal/semantic/literal_expander.cr#L508

Please add `else` here like `"\<\%(when\|else\)\>[?!]\@!"`

**Review by rhysd — CHANGES_REQUESTED** [review]

**Comment by rhysd** [comment]
@sudo-nice Thank you for sending this patch. Please take a look at my review comment above. (Please ignore CI result. )

**Comment by sudo-nice** [comment]
Thank you got the remark, done.

**Review by rhysd — APPROVED** [review]

**Comment by rhysd** [comment]
Thank you

**Pull request merged**